irods/manager/user_manager.py

```diff
@@ -134,6 +134,10 @@ def modify_password(self, old_value, new_value, modify_irods_authentication_file
         """
         with self.sess.pool.get_connection() as conn:
 
+            if old_value != self.sess.pool.account.password or not isinstance(new_value, str) \
```

Comment:
Here, we are just taking the  old_value of the password provided in the call and comparing it to the password used to authenticate the account to the server.  I realize there is some redundancy there, but I think it's better this way because:
   - it's consistent with `ipasswd` behavior
   - it helps to prevent someone  from calling modify_password "accidentally" on their own login or (in the admin's case) on a user whose login he didn't mean to change.

Reply:
I see. `self.sess.pool.account.password` is the active account's password.

This one condition effectively guards against unnecessary network calls.